sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/util/IdentitySSLUtil.java

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.implementation.util;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import java.security.KeyManagementException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+public class IdentitySSLUtil {
+    public static final HostnameVerifier ALL_HOSTS_ACCEPT_HOSTNAME_VERIFIER;
+
+    static {
+        ALL_HOSTS_ACCEPT_HOSTNAME_VERIFIER = new HostnameVerifier() {
+            @SuppressWarnings("BadHostnameVerifier")
+            @Override
+            public boolean verify(String hostname, SSLSession session) {
+                return true;
+            }
+        };
+    }
+
+    /**
+     *
+     * Pins the specified HTTPS URL Connection to work against a specific server-side certificate with
+     * the specified thumbprint only.
+     *
+     * @param className The class calling the method.
+     * @param httpsUrlConnection The https url connection to configure
+     * @param certificateThumbprint The thumbprint of the certificate
+     */
+    public static void addTrustedCertificateThumbprint(String className, HttpsURLConnection httpsUrlConnection,
+                                                       String certificateThumbprint) {
+
+        ClientLogger logger = new ClientLogger(className);
```

Comment:
This can be converted to a static logger.

Reply:
We have a checkstyle rule to not have static loggers.

Reply:
For static helper classes, it's okay to use static logger and add checkstyle suppression.